controllers/etcdpeer_controller.go

```diff
@@ -16,14 +20,104 @@ type EtcdPeerReconciler struct {
 	Log logr.Logger
 }
 
+func ignoreNotFound(err error) error {
+	if apierrs.IsNotFound(err) {
+		return nil
+	}
+	return err
+}
+
+const (
+	EtcdImage = "quay.io/coreos/etcd:v3.2.27"
+)
+
 // +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdpeers,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdpeers/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=replicaset,verbs=get;update;patch;create
+
+func defineReplicaSet(etcdPeer etcdv1alpha1.EtcdPeer) (appsv1.ReplicaSet, error) {
+	var replicas int32 = 1
+
+	return appsv1.ReplicaSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:      make(map[string]string),
+			Annotations: make(map[string]string),
+			Name:        etcdPeer.Name,
+			Namespace:   etcdPeer.Namespace,
+		},
+		Spec: appsv1.ReplicaSetSpec{
+			// This will *always* be 1. Other peers are handled by other EtcdPeers.
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app":  "etcd",
+					"peer": etcdPeer.Name,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app":  "etcd",
+						"peer": etcdPeer.Name,
+					},
+					Annotations: make(map[string]string),
+					Name:        etcdPeer.Name,
+					Namespace:   etcdPeer.Namespace,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "etcd",
+							Image: EtcdImage,
+						},
+					},
+				},
+			},
+		},
+	}, nil
+}
 
 func (r *EtcdPeerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("etcdpeer", req.NamespacedName)
+	ctx := context.Background()
+	log := r.Log.WithValues("etcdpeer", req.NamespacedName)
+
+	var etcdPeer etcdv1alpha1.EtcdPeer
+	if err := r.Get(ctx, req.NamespacedName, &etcdPeer); err != nil {
+		log.Error(err, "unable to fetch EtcdPeer")
+		return ctrl.Result{}, ignoreNotFound(err)
+	}
+
+	log.V(1).Info("Found EtcdPeer",
+		"name", etcdPeer.Name,
+		"bootstrapPeers", etcdPeer.Spec.BootstrapPeers)
+
+	var existingReplicaSet appsv1.ReplicaSet
+	err := r.Get(ctx,
+		client.ObjectKey{
```

Comment:
I think you can just use req.NamespacedName here

Reply:
Because the replica set happens to have the same name as the `EtcdPeer`? I wonder if it would be best to generate my proposed `ReplicaSet` (using `defineReplicaSet`, which is a pure function) and then get it's name. So we're only defining the name of the replica set in one place and it won't break if we change that conversion.

Reply:
Sure :smile: this is likely to change in future anyway, nothing to worry about now